### ISSUE TITLE
unexpected warning from clang

### ISSUE THREAD

**Issue by BaoshanPang** [llvm/llvm-project#75984]
Can someone tell me why clang report a warning here? isn't lludump.a a type of "unsigned long long"? Thanks.

```c
#include <stdio.h>

typedef unsigned char u8;
typedef unsigned char u08;
typedef unsigned short u16;
typedef unsigned int u32;
typedef unsigned long long u64;

struct llu_dump_test {
  u64 a : 4;
  u64 b : 56;
  u64 c : 4;
};

int main() {
  struct llu_dump_test lludump;

  printf("%llu\n", lludump.a);

  return 0;
}
```

bpang@US-BPANG:~$ clang -c e.c
```console
e.c:18:20: warning: format specifies type 'unsigned long long' but the argument has type 'int' [-Wformat]
  printf("%llu\n", lludump.a);
          ~~~~     ^~~~~~~~~
          %d
1 warning generated.
```

**Comment by shafik** [comment]
Note that gcc agrees with clang here: https://godbolt.org/z/Gva79oa6d

This is correct as far as I can tell b/c `a` has a width of `4` and therefore `int` can represent all of its values. See `6.3.1.1 Boolean, characters, and integers`:

> If an int can represent all values of the original type (as restricted by the width, for a
bit-field), the value is converted to an int; otherwise, it is converted to an unsigned
int. These are called the integer promotions

CC @AaronBallman

**Comment by BaoshanPang** [comment]
@shafik  Thanks. That's the answer I am looking for.
